src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs

```diff
@@ -389,10 +395,24 @@ public void TrySync()
             this.logger.LogTrace("(-)");
         }
 
-        private ChainedBlock GetPendingTipOrChainTip()
+        /// <summary>
+        /// Determines if the peer's headers are synced with ours.
+        /// </summary>
+        /// <remarks>
+        /// It is possible that peer is in IBD even though he has all the headers so we can't assume with 100% certainty that peer is fully synced.
+        /// </remarks>
+        public bool IsSynced()
+        {
+            if (this.pendingTip == null)
+                return false;
+
+            return (this.pendingTip.Height >= this.chainState.ConsensusTip.Height &&
+                    this.pendingTip.FindAncestorOrSelf(this.chainState.ConsensusTip) != null);
+        }
+
+        public ChainedBlock GetPendingTipOrChainTip()
         {
-            this.pendingTip = this.pendingTip ?? this.chainState.ConsensusTip ?? this.Chain.Tip;
-            return this.pendingTip;
+            return this.pendingTip ?? this.chainState.ConsensusTip ?? this.Chain.Tip;
```

Comment:
eh?

Reply:
1) 'GetPendingTipOrChainTip' - the name of the function tells that it will return the value, not modify it. It used to have a logical error. 
2) By setting the tip to our tip we make a false image of the peer's tip.

Reply:
This has some serious consequences, I did not really expect this to be part of this PR. Need to think about all changes you made again now ...

Reply:
OK, I agree the original code did have that misleading logic, I just did not expect it to be fixed on this PR, that what confused me everywhere.

So if we are about to fix this logic in this PR, we should really be strict (including that in ConnectionManager console display). So I will review again with this in mind that we want to fix that logic.

Reply:
OK, if we want to fix that, probably this method should be deleted completely